### PR TITLE
`Modeling exercises`: Improve modeling editor readonly view

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context
Apollon UML Diagrams have been shown overcomplicated in their readonly views such as in modeling managament detail pages or as example solution in the students view.

### Description
The UML Diagram is in readonly now exported as SVG element and will be displayed on plain background to reduce complexity.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Create a modeling exercise with due date and example solution set
3. View the exercise in the management detail view
4. View the example solution (once the example solution publication date is over) in the students view
5. Verify that it looks correct

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
image
image

## Summary by CodeRabbit

- **New Features**
	- Enhanced the modeling editor with readonly mode support, including the ability to view diagrams as SVGs.
- **Style**
	- Improved styling for text elements within `.readonly-diagram` to enhance visual contrast.
- **Refactor**
	- Updated conditional rendering logic in the modeling editor for a more intuitive user experience.

### CODE REVIEW

File: src/main/webapp/app/exercises/modeling/shared/modeling-editor.component.ts [hunk 1]
```diff
@@ -34,9 +35,13 @@ export class ModelingEditorComponent extends ModelingComponent implements AfterV
     mouseDownListener: ((this: Document, ev: MouseEvent) => any) | undefined;
     scrollListener: ((this: Document, ev: Event) => any) | undefined;
 
+    readonlyApollonDiagram?: SVG;
+    readOnlySVG: any;
```
Comment (matthiaslehnertum) [inline src/main/webapp/app/exercises/modeling/shared/modeling-editor.component.ts:39]: Does the type here have to be 'any'?

**Comment by loreanvictor** [comment]
@milljoniaer quick question: how about diagrams with some feedback/assessment attached? are they considered as readonly as well? if so, will the feedback/assessment be displayed and/or will it highlight the corresponding elements?

File: src/main/webapp/app/exercises/modeling/shared/modeling-editor.component.ts [hunk 2]
```diff
@@ -46,15 +51,24 @@ export class ModelingEditorComponent extends ModelingComponent implements AfterV
...
      * If resizeOptions is set to true, resizes the editor according to interactions.
      */
-    ngAfterViewInit(): void {
+    async ngAfterViewInit(): Promise<void> {
         this.initializeApollonEditor();
-        this.guidedTourService.checkModelingComponent().subscribe((key) => {
-            if (key) {
-                this.assessModelForGuidedTour(key, this.getCurrentModel());
+        if (this.readOnly) {
+            console.log(this.apollonEditor);
```
Comment (loreanvictor) [inline src/main/webapp/app/exercises/modeling/shared/modeling-editor.component.ts:57]: I suppose this log shouldn't be here?

File: src/main/webapp/app/exercises/modeling/shared/modeling-editor.component.ts (hunk 1 above)
Comment (milljoniaer) [reply src/main/webapp/app/exercises/modeling/shared/modeling-editor.component.ts]: ups, overlooked 😅

**Review by matthiaslehnertum — APPROVED** [review]

**Review by rstief — DISMISSED** [review]
code 👍

**Review by coolchock — APPROVED** [review]

**Review by pzdr7 — COMMENTED** [review]
Tested on TS2. Light mode looks fine, but I don't suppose you could do anything about the font color when the user is in dark mode? It's a bit hard to read imo

<details>
  <summary>Editing in dark mode (unchanged)</summary>

grafik
</details>

<details>
  <summary>Viewing the example solution in the detail view</summary>

grafik

</details>

**Review by loreanvictor — APPROVED** [review]
code looks good to me.

I think an improvement (that can be done later?) would be to do the exporting in `ngAfterViewChecked()` instead, so that if input of the component changes the rendered diagram updates as well.

**Review by rstief — APPROVED** [review]

**Review by coolchock — DISMISSED** [review]
re-approve

**Review by pzdr7 — DISMISSED** [review]
Tested locally, looks good in dark and light mode 👍

**Review by matthiaslehnertum — DISMISSED** [review]

**Pull request merged**